src/Store/SearchStoreConnector.re

```diff
@@ -29,20 +29,25 @@ let start = () => {
   };
 
   let updater = (state: Model.State.t, action) => {
-    let searchPane =
-      switch (action) {
-      | SearchShow => Some(Model.Search.initial)
-
-      | SearchHide => None
-
-      | _ =>
-        switch (state.searchPane) {
-        | Some(searchPane) => Some(searchUpdater(searchPane, action))
-        | None => None
-        }
-      };
-
-    ({...state, searchPane}, Isolinear.Effect.none);
+    switch (action) {
+    | Tick(_) => (state, Isolinear.Effect.none)
```

Comment:
I just spent a few hours learning that I need this "hack" to prevent `TickTest` from failing. I understand it's needed because there's a physical equality check somewhere that prevents re-rendering if the state hasn't actually changed, and without this it would re-render on every tick. Which is bad. But is the `Tick` really needed anymore? Can these use cases be solved using the new animation hooks and subscriptions instead?

Reply:
> I just spent a few hours learning that I need this "hack" to prevent TickTest from failing.

Ouch, sorry @glennsl ! 

>  I understand it's needed because there's a physical equality check somewhere that prevents re-rendering if the state hasn't actually changed, and without this it would re-render on every tick. Which is bad.

This is exactly right - don't want to re-render on every `Tick`.

> But is the Tick really needed anymore? Can these use cases be solved using the new animation hooks and subscriptions instead?

I think we could remove it (and it would be ideal, so we don't need the complexity of having our reducers by resilient about ticks). 

The main places that we still use it are:
- `SyntaxHighlightingStoreConnector` - used to execute periodic syntax highlighting jobs w/ time slicing. This could go away with #922  - since we'll be deferring that work to a separate process running full-tilt (we'd get responses from the new syntax highlight process).
- `HoverStoreConnector` - this was used for some animation, but could go away with a refactoring to use the new animation hooks
- `KeyDisplayerConnector` - this uses the `Tick` to expire some events when active. But we could refactor the API such that this isn't needed.
- `ExtensionClientStoreConnector` - it uses the `Tick` to `pump` the Node RPC client. This could be refactored to use a 'scheduler' instead.

It would be great to remove that and not have that 'gotcha' around.

Reply:
Awesome! I'll add that to the "Technical" project.